pkg/distributor/errors.go

```diff
@@ -158,6 +159,37 @@ func (e requestRateLimitedError) errorCause() mimirpb.ErrorCause {
 // Ensure that requestRateLimitedError implements distributorError.
 var _ distributorError = requestRateLimitedError{}
 
+// ingesterPushError is an error used to represent a failed attempt to push to the ingester.
+type ingesterPushError struct {
+	message string
+	cause   mimirpb.ErrorCause
+}
+
+// newIngesterPushError creates a ingesterPushError error representing the given status object.
+func newIngesterPushError(stat *status.Status) ingesterPushError {
+	errorCause := mimirpb.INVALID
+	if len(stat.Details()) == 1 {
```

Comment:
I assume it's not possible for the size of the slice returned by `.Details()` to be bigger than a single element? Should we handle that anyway?

Reply:
It is possible for a `status.Status` to have more than one proto message returned by `Details()`. Nevertheless, errors implementing both `ingesterError` and `distributorError` interfaces are supposed to have exactly one error detail of type `mimirpb.WriteErrorDetails`. This is why the check is performed: if it is not satisfied, the default cause `mimirpb.INVALID` is assigned to this error.